### PR TITLE
This makes the scripts executable on a fresh R installation.

### PR DESCRIPTION
**Pull request by lemire**
This may not be a good idea, but for someone like myself who only use R from time-to-time, and often on a fresh R setup, having self-contained scripts that install their own dependencies is quality-of-life issue.

### CODE REVIEW

**Comment by eddelbuettel** [comment]
About to go for a (belated) run and will clean the commit up a little later.  These days the recommended idiom is

```r
if (!requireNamespace("somepackage", quietly=TRUE) doSomethingHere()
```

It is a matter of taste if make scripts install their requirements.  I used to do it a little more ~20 years ago, I now find it (mostly) redundant as we a) do work with packages anyway and b) packages take care of this.

It is definitely useful for _optional_ packages having a `Suggests:` in `DESCRIPTION`.

Lastly, compulsory advertisement:  You want probably want [littler](https://github.com/eddelbuettel/littler) to give you `r` (at least on real computers, those infected by some Cupertino-based disease need to alias to `lr` or some such as the brain surgeons over there default to `r` being equal to `R`.  Grrrrrrr.)

File: README.md [hunk 1]
```diff
@@ -26,6 +26,10 @@ QCon](http://www.youtube.com/watch?v=wlvKAT7SZIQ) (voted best talk).
 ### Example
 
 ```r
+if (!require("RcppSimdJson")) install.packages("RcppSimdJson")
```
Comment (eddelbuettel) [inline README.md]: On balance, no.

In a README.md conciseness trumps completeness.  It is also the top-level README _for the package_.  We will assume people can install it.

File: demo/simpleBenchmark.R [hunk 2]
```diff
@@ -1,6 +1,14 @@
+#!/usr/bin/env Rscript
 
+if (!require("microbenchmark")) install.packages("microbenchmark")
```
Comment (eddelbuettel) [inline demo/simpleBenchmark.R]: Or simply a `stopifnot()` which now has customizable errors. 

Auto-installation is no panacea either as not every user will have a compiler.

File: demo/simpleBenchmark.R [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
 library(microbenchmark)
 
+if (!require("jsonify")) install.packages("jsonify")
+if (!require("jsonlite")) install.packages("jsonlite")
+if (!require("RcppSimdJson")) install.packages("RcppSimdJson")
+if (!require("ndjson")) install.packages("ndjson")
+if (!require("RJSONIO")) install.packages("RJSONIO")
```
Comment (eddelbuettel) [inline demo/simpleBenchmark.R]: This makes some sense here and I will clean that up later.

**Comment by lemire** [comment]
This PR can be thrown away without hurting my feelings at all. ;-)

**Comment by eddelbuettel** [comment]
I added a small amount of spit and polish. Let me know what you think.

**Comment by lemire** [comment]
I think that this is much better!!! I recommend merging this.

**Pull request merged**